### PR TITLE
Bit of reference handling for custom food holders

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

`ingredients` seems to hold a reference to all the food in our atom's contents that are used to make up our custom food, but it doesn't clear it anywhere.

So I implemented `Exited`, as well as cutting the list on component `Destroy`.

### CODE REVIEW

File: code/datums/components/customizable_reagent_holder.dm [hunk 1]
```diff
@@ -289,3 +291,9 @@
 	context[SCREENTIP_CONTEXT_LMB] = "[screentip_verb] [held_item]"
 
 	return CONTEXTUAL_SCREENTIP_SET
+
+/// Clear refs if our food goes away
+/datum/component/customizable_reagent_holder/proc/food_exited(datum/source, atom/movable/gone)
+	SIGNAL_HANDLER
+	if(gone in ingredients)
+		LAZYREMOVE(ingredients, gone)
```
Comment (LemonInTheDark) [inline code/datums/components/customizable_reagent_holder.dm:299]: why the in check, if it's not it'll do nothing (in the same time cause they're both O(n)

File: code/datums/components/customizable_reagent_holder.dm [hunk 2]
```diff
@@ -92,18 +95,17 @@
...
+		var/obj/item/ingredient = ingredients[i]
+		var/ending = ", "
+		switch(length(ingredients))
+			if (2)
+				if (i == 1)
+					ending = " and "
+			if (3 to INFINITY)
+				if (i == LAZYLEN(ingredients) - 1)
+					ending = ", and "
+		ingredients_listed += "\a [ingredient.name][ending]"
```
Comment (LemonInTheDark) [inline code/datums/components/customizable_reagent_holder.dm:108]: this code makes me very depressed (this isn't hot right? would it be useful to handle as a list instead?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/datums/components/customizable_reagent_holder.dm (hunk 1 above)
Comment (MrMelbert) [reply code/datums/components/customizable_reagent_holder.dm:299]: oh yeah

File: code/datums/components/customizable_reagent_holder.dm (hunk 2 above)
Comment (MrMelbert) [reply code/datums/components/customizable_reagent_holder.dm:108]: I actually have no idea why this doesn't just use english_list

Comment (LemonInTheDark) [reply code/datums/components/customizable_reagent_holder.dm:108]: the \a maybe?

**Review by LemonInTheDark — APPROVED** [review]

**Pull request merged**